app/lib/methods/updateMessages.js

```diff
@@ -58,6 +58,25 @@ export default function updateMessages({ rid, update = [], remove = [], loaderIt
 			// filter loaders to delete
 			let loadersToDelete = allMessagesRecords.filter(i1 => update.find(i2 => i1.id === generateLoadMoreId(i2._id)));
 
+			// Delete
```

Comment:
Moved delete to top, because I found a couple of issues on Bugsnag complaining about trying to delete a record with pending updates.
So let's schedule to delete first and handle the issue on update block.
Of course it doesn't solve the issue, but the issue itself is a matter of timing and might be fixed only with JSI (and its instant changes).